lib/puppet_x/puppetlabs/aws.rb

```diff
@@ -98,6 +98,12 @@ def self.tags_for(item)
         tags
       end
 
+      def target_region
+        target = resource ? resource[:region] || region : region
```

Comment:
region in this context is the value of the region property. This basically says:

* If you specify a region (ie. on the command line like `puppet resource ec2_securitygroup bob region=blah` or in the DSL, use that
* If that isn't present, fall back to the value of region for the resource